lms/static/scripts/frontend_apps/components/BookPicker.tsx

```diff
@@ -69,43 +69,53 @@ export default function BookPicker({
   const canSubmit =
     (step === 'select-book' && book) || (step === 'select-chapter' && chapter);
 
-  // Provide a spacious modal UI when listing chapters (taller modal)
-  const showChapters = step === 'select-chapter' && !error;
+  /**
+   * Prevent submission of content form, which can raise an error because
+   * content has not been selected.
+   */
+  const onCancelButtonClick = (event: Event) => {
+    event.preventDefault();
+    event.stopPropagation();
+    onCancel();
+  };
 
   return (
     <Modal
       // Opt out of Modal's automatic focus handling; route focus manually in
       // sub-components
-      initialFocus={null}
-      onCancel={onCancel}
-      contentClass={classnames('LMS-Dialog LMS-Dialog--wide', {
-        'LMS-Dialog--tall': showChapters,
-      })}
+      initialFocus={'manual'}
+      onClose={onCancel}
       title={
         step === 'select-book'
           ? 'Paste link to VitalSource book'
           : 'Pick where to start reading' // "Select a chapter"
       }
-      buttons={[
-        <LabeledButton
-          key="submit"
-          data-testid="select-button"
-          disabled={!canSubmit}
-          onClick={() => {
-            // nb. The `book` and `chapter` checks should be redundant, as the button
-            // should not be clickable if no book/chapter is selected, but they
-            // keep TS happy.
-            if (step === 'select-book' && book) {
-              confirmBook(book);
-            } else if (step === 'select-chapter' && chapter) {
-              confirmChapter(chapter);
-            }
-          }}
-          variant="primary"
-        >
-          {step === 'select-book' ? 'Select book' : 'Select'}
-        </LabeledButton>,
-      ]}
+      width="lg"
+      buttons={
+        <>
+          <Button data-testid="cancel-button" onClick={onCancelButtonClick}>
+            Cancel
+          </Button>
+          <Button
+            key="submit"
+            data-testid="select-button"
+            disabled={!canSubmit}
+            onClick={() => {
+              // nb. The `book` and `chapter` checks should be redundant, as the button
+              // should not be clickable if no book/chapter is selected, but they
+              // keep TS happy.
```

Comment:
Perhaps this can be replaced by

```ts
if (!canSubmit) {
  return;
}
```

I guess TS is not smart enough to detect the fact that the button won't be enabled, but if the check is done inside the same function, it should work as a type guard.

But this is mostly me thinking out loud. I don't think it's a big improvement, so I'm ok with the way it already is. Specially considering this code was already like this.